cpp/Integri/XsdComparator/XsdComparator.cpp

```diff
@@ -0,0 +1,219 @@
+#include "XsdComparator.h"
+#include <iostream>
+#include <stdio.h>
+#include <tchar.h>
+#include <windows.h>
+#include <atlbase.h>
+#include <comutil.h>
+
+using namespace std;
+
+XsdComparator::XsdComparator(Reporter* reporter)
+{
+	this->reporter = reporter;
+}
+
+void XsdComparator::Compare(const std::string file1, const std::string file2)
+{
+	try
+	{
+		MSXML2::IXMLDOMElement* firstRoot;
+		MSXML2::IXMLDOMElement* secondRoot;
+
+		GetRootElement(file1, &firstRoot);
+		GetRootElement(file2, &secondRoot);
+
+		Compare(firstRoot, secondRoot);
+	}
+	catch (const std::exception& e)
+	{
+		printf(e.what());
+	}
+}
+
+void XsdComparator::Compare(MSXML2::IXMLDOMElement* firstRoot, MSXML2::IXMLDOMElement* secondRoot)
+{
+	std::string firstRootElementName = GetElementName((MSXML2::IXMLDOMNode*)firstRoot);
+	std::string secondRootElementName = GetElementName((MSXML2::IXMLDOMNode*)secondRoot);
+
+	std::string firstRootName = GetValueOfNameAttribute((MSXML2::IXMLDOMNode*)firstRoot);
+	std::string secondRootName = GetValueOfNameAttribute((MSXML2::IXMLDOMNode*)secondRoot);
+
+	if (firstRootName.compare(secondRootName) == 0)
+	{
+		reporter->Add(None, 0, firstRootElementName, firstRootName);
+
+		MSXML2::IXMLDOMNodeList* firstChildren;
+		MSXML2::IXMLDOMNodeList* secondChildren;
+
+		firstRoot->get_childNodes(&firstChildren);
+		secondRoot->get_childNodes(&secondChildren);
+
+		CompareChildren(firstChildren, secondChildren, 1);
+
+		firstChildren->Release();
+		secondChildren->Release();
+	}
+	else
+	{
+		MessageBox(0, _T("Root element names does not match"), _T("Error"), MB_OK);
+	}
+}
+
+std::map<std::string, MSXML2::IXMLDOMNode*> XsdComparator::GetNodesMap(MSXML2::IXMLDOMNodeList* nodes)
```

Comment:
I wonder, this is not yet my strength either, if a r-value reference could be used here to avoid copying the whole std::map.
`std::map<std::string, MSXML2::IXMLDOMNode*>&& XsdComparator::GetNodesMap(MSXML2::IXMLDOMNodeList* nodes)`